### DISCUSSION TITLE
ShaderMaterial: "Class constructors cannot be invoked without 'new'"

### DISCUSSION THREAD

**Discussion by molp** [pmndrs/drei#516] [Q&A] [answered]
Hello,

I am trying to create a custom shader with shaderMaterial, but cannot get it to run. Here is a minimal Sandbox 

https://codesandbox.io/s/drei-shader-material-9sk5d?file=/src/App.js

The error I get:

> Class constructors cannot be invoked without 'new'

Any idea what I am doing wrong?

**Comment by joshuaellis** [comment, marked as answer]
L40 should be `lineShaderMaterial` not `LineShaderMaterial`.

**Reply by joshuaellis** [reply]
it's not a component, so you need to lowercase the first letter because you've added it to the catalogue of jsx components in the reconciler 😄

**Reply by molp** [reply]
Ok, now I feel stupid :D Thank you very much for the quick help & explanation!
